### PR TITLE
docs(testing): behavior driven development and mocking

### PR DESCRIPTION
**Pull request by KyleJune**
I created these new pages from the documentation in https://deno.land/std@0.135.0/testing/README.md. The only difference is one less `#` for each heading.

### CODE REVIEW

**Comment by KyleJune** [comment]
I missed adding the new pages to `toc.json`. My last commit adds the new pages and I verified they show up in the preview deployment.

File: testing/behavior_driven_development.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
+# Behavior-driven development
+
+With the `bdd.ts` module you can write your tests in a familiar format for
```
Comment (bartlomieju) [inline testing/behavior_driven_development.md:3]: Maybe add a link to the module here?

File: testing/behavior_driven_development.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+The `describe` function creates a block that groups together several related
+tests. The `it` function registers an individual test case.
+
+## Hooks
+
+There are 4 types of hooks available for test suites. A test suite can have
+multiples of each type of hook, they will be called in the order that they are
+registered. The `afterEach` and `afterAll` hooks will be called whether or not
+the test case passes. The *All hooks will be called once for the whole group
```
Comment (bartlomieju) [inline testing/behavior_driven_development.md:15]:
```suggestion
the test case passes. The `*All` hooks will be called once for the whole group
```

File: testing/behavior_driven_development.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+while the *Each hooks will be called for each individual test case.
```
Comment (bartlomieju) [inline testing/behavior_driven_development.md:16]:
```suggestion
while the `*Each` hooks will be called for each individual test case.
```

**Review by bartlomieju — COMMENTED** [review]
Great, just a few minor nitpicks

**Comment by KyleJune** [comment]
I've addressed all the issues.

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @KyleJune!

**Pull request merged**